File: 2024/week51-60/week55/409_1715.py
import sys
import heapq
input=sys.stdin.readline

n=int(input())
card_set=[int(input()) for _ in range(n)]
heapq.heapify(card_set)
result=0
while len(card_set)>1:
    card1=heapq.heappop(card_set)
    card2=heapq.heappop(card_set)
    result+=card1+card2
    heapq.heappush(card_set,card1+card2)
print(result)

File: 2024/week51-60/week55/409_1715.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));

        int n=Integer.parseInt(br.readLine());
        PriorityQueue<Integer> card_set= new PriorityQueue<Integer>();
        for (int i = 0; i < n; i++) {
            int card=Integer.parseInt(br.readLine());
            card_set.offer(card);
        }
        int result=0;
        while(card_set.size()>1){
            int card1=card_set.poll();
            int card2=card_set.poll();
            result+=card1+card2;
            card_set.offer(card1+card2);
        }
        System.out.println(result);


    }
}